text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1179 @@
+# AWS DynamoDB Global Table L2 Construct
+
+* **Original Author(s):**: @vinayak-kukreja
+* **Tracking Issue**: #510
+* **API Bar Raiser**: @rix0rrr
+
+Users will now be able to replicate their DynamoDB table to multiple regions using the Global Table L2 construct.
+This feature will be using the CloudFormation resource for global table and users will no longer need to rely
+custom resources for provisioning global tables.
+
+## Working Backwards
+
+The following is ReadMe for DynamoDB Global Table.
+
+__NOTE:__ This just includes properties that are different from the Table construct. For an in detailed comparision
+between properties, take a look at the [appendix](#difference-between-tableprops-and-globaltableprops).
+
+### DynamoDB Global Table
+
+[DynamoDB Global Table](https://aws.amazon.com/dynamodb/global-tables/) lets you provision a table that can be
+replicated across different regions. It can also be deployed to just one region and will cost the same as a
+single DynamoDB table.
+
+It is also multi-active database, that means there is no primary table and all the tables created as called
+as replicas and all replicas support reads and writes. Writes to a replica are eventually propagated to other
+replicas where conflicts are resolved by 'last writer wins'.
+
+#### Read and Write Capacity
+
+Global tables by default have `BillingMode.PAY_PER_REQUEST`. If you choose the `BillingMode.PROVISIONED` mode,
+then you will need to specify read and write capacities. If these values are specified at global table level, then
+those values are used for each replica and its global secondary indexes(GSIs).
+
+```typescript
+new GlobalTable(tableStack, 'GlobalTable', {
+  tableName: 'FooTable',
+  partitionKey: {
+    name: 'FooHashKey',
+    type: AttributeType.STRING,
+  },
+  billingMode: BillingMode.PROVISIONED,
+  write: Capacity.autoscaled({ max: 70 }),
+  read: Capacity.fixed(20),
```

Comment:
I see, taking a look.

Reply:
This makes sense to me. One of the reason for keeping it the same was to keep it similar to the table API to minimize user changes. But this would not be that big of a churn to update.

It does make sense, since it is a stronger contract and we do not need `read` and `write` values for `BillingMode.ondemand` which the user can still pass in with the API I had proposed. That can be validated and thrown an error for, but what you are proposing would take away the need for it. :)